libs/langchain/langchain/memory/chat_message_histories/rocksetdb.py

```diff
@@ -0,0 +1,258 @@
+                        LIMIT 1
+                    """,
+                    session_id=self.session_id,
+                    message_id=message_id,
+                ),
+            )
+            != 0,
+            RocksetChatMessageHistory.ADD_TIMEOUT_MS,
+            message_id=message_id,
+        )
+
+    def _create_empty_doc(self) -> None:
+        """Creates or replaces a document for this message history with no
+        messages"""
+        self.client.Documents.add_documents(
+            collection=self.collection,
+            workspace=self.workspace,
+            data=[{"_id": self.session_id, self.messages_key: []}],
+        )
+
+    def __init__(
+        self,
+        session_id: str,
+        client: Any,
+        collection: str,
+        workspace: str = "commons",
+        messages_key: str = "messages",
+        sync: bool = False,
+        message_uuid_method: Callable[[], Union[str, int]] = lambda: str(uuid4()),
+    ) -> None:
+        """Constructs a new RocksetChatMessageHistory.
+
+        Args:
+            - session_id: The ID of the chat session
+            - client: The RocksetClient object to use to query
+            - collection: The name of the collection to use to store chat
+                          messages. If a collection with the given name
+                          does not exist in the workspace, it is created.
+            - workspace: The workspace containing `collection`. Defaults
+                         to `"commons"`
+            - messages_key: The DB column containing message history.
+                            Defaults to `"messages"`
+            - sync: Whether to wait for messages to be added. Defaults
+                    to `False`. NOTE: setting this to `True` will slow
+                    down performance.
+            - message_uuid_method: The method that generates message IDs.
+                    If set, all messages will have an `id` field within the
+                    `additional_kwargs` property. If this param is not set
+                    and `sync` is `False`, message IDs will not be created.
+                    If this param is not set and `sync` is `True`, the
+                    `uuid.uuid4` method will be used to create message IDs.
+        """
+        try:
+            import rockset
+        except ImportError:
+            raise ImportError(
+                "Could not import rockset client python package. "
+                "Please install it with `pip install rockset`."
+            )
+
+        if not isinstance(client, rockset.RocksetClient):
+            raise ValueError(
+                f"client should be an instance of rockset.RocksetClient, "
+                f"got {type(client)}"
+            )
+
+        self.session_id = session_id
+        self.client = client
+        self.collection = collection
+        self.workspace = workspace
+        self.location = f""""{self.workspace}".{self.collection}"""
```

Comment:
double checking, this is meant to have double quotes around self.workspace?

Reply:
That is correct. Actually, it would be better if `{self.collection}` had quotes around it too, just in case if a collection has special characters in its name. I fixed this in https://github.com/langchain-ai/langchain/pull/8940/commits/76a9d8c4e9cbf5818028c32450e8f88efd4338c1 and made the quotes more readable.